recipe/conda_build_config.yaml

```diff
@@ -46,6 +46,10 @@ zip_keys:
 pin_run_as_build:
   arpack:
     max_pin: x.x
+  boost:
+    max_pin: x.x.x
+  boost_cpp:
```

Comment:
Isn't the package called "boost-cpp"?

Reply:
Yeah, but yaml doesn't allow that so it's translated. Also see `r_base`.

Reply:
I thought you could do something like:

```
"boost-cpp":
    max_pin: x.x.x
```

Reply:
Hmm, maybe, and that would be nicer. Not sure how that works, can investigate later....

Reply:
Think we have to use `_`. The parser got messed up with `-` for me previously.

Reply:
@jakirkham - which parser? Just curious, because it seems to me that using the actual package names is ultimately the right way to go.  If it is breaking I imagine this is an issue in upstream conda or conda-build

Reply:
It's by design. Keys in the config yaml are exposed as jinja variables in the meta.yaml and env variables in the build process, both of which require the name not having a `-`

Reply:
oof that is annoying.  That means that `-` and `_` should always be interchangeable, then

Reply:
I think `conda-build` when trying to re-render a feedstock.

TBH don't think this is that bad in practice. Don't think we have packages that differ only by `-`/`_` and if we do we are bound to run into tons of issues from our users, which is incentive enough to steer clear.

Reply:
@isuruf described it correctly.  - and _ are treated as equivalent for the purposes of matching package names to variable names (the foundation of what gets looped over).

If this is not desirable, you can make the key in CBC.yaml any arbitrary string, and then use it explicitly wherever you need it.

Reply:
Looks like @scopatz was right. If it's a top level key you need `_`, but for `pin_run_as_build` you need the exact package name. Very confusing. https://github.com/conda-forge/postgis-feedstock/pull/8/files/#r187479333

Reply:
That is an annoying detail that grew out of pin_run_as_build being implemented completely differently (and in a different part of the source) from run_exports and the jinja2 functions.  PRs to fix it would be welcome.  Sorry you guys have been the pin_run_as_build guinea pigs.  That's a feature that we just haven't really used much in defaults.  With r-base, it was easy enough for us to hack around.  With all of your packages, it is a much bigger problem.